torch/csrc/autograd/engine.cpp

```diff
@@ -931,6 +935,7 @@ auto Engine::start_threads() -> void {
   thread_pool_shared_ = std::make_shared<ThreadPoolShared>();
 
   for (int i = 0; i < num_threads; ++i) {
+    non_reentrant_thread_count_.fetch_add(1);
```

Comment:
You don't really need an atomic additions for each one here. You can just set it to `num_threads` above.
This function is not expected to be thread safe as it only ever called inside `std::call_once()`.

Reply:
Updated